extensions/oidc-client-registration/deployment/src/main/java/io/quarkus/oidc/client/registration/deployment/devservices/keycloak/KeycloakDevServiceRequiredBuildStep.java

```diff
@@ -0,0 +1,93 @@
+package io.quarkus.oidc.client.registration.deployment.devservices.keycloak;
+
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.logging.Logger;
+import org.keycloak.common.util.MultivaluedHashMap;
+import org.keycloak.representations.idm.ComponentExportRepresentation;
+import org.keycloak.representations.idm.RealmRepresentation;
+
+import io.quarkus.deployment.IsDevelopment;
+import io.quarkus.deployment.IsNormal;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.BuildSteps;
+import io.quarkus.deployment.builditem.DockerStatusBuildItem;
+import io.quarkus.deployment.dev.devservices.GlobalDevServicesConfig;
+import io.quarkus.devservices.keycloak.KeycloakAdminPageBuildItem;
+import io.quarkus.devservices.keycloak.KeycloakDevServicesConfigurator;
+import io.quarkus.devservices.keycloak.KeycloakDevServicesRequiredBuildItem;
+import io.quarkus.devui.spi.page.CardPageBuildItem;
+import io.quarkus.oidc.client.registration.deployment.OidcClientRegistrationBuildStep;
+import io.quarkus.runtime.configuration.ConfigUtils;
+
+@BuildSteps(onlyIfNot = IsNormal.class, onlyIf = { OidcClientRegistrationBuildStep.IsEnabled.class,
+        GlobalDevServicesConfig.Enabled.class })
+public class KeycloakDevServiceRequiredBuildStep {
+
+    private static final Logger LOG = Logger.getLogger(KeycloakDevServiceRequiredBuildStep.class);
+    private static final String OIDC_CLIENT_REG_AUTH_SERVER_URL_CONFIG_KEY = "quarkus.oidc-client-registration.auth-server-url";
+    private static final String OIDC_AUTH_SERVER_URL_CONFIG_KEY = "quarkus.oidc.auth-server-url";
+
+    @BuildStep
+    void requireKeycloakDevService(BuildProducer<KeycloakDevServicesRequiredBuildItem> keycloakDevSvcRequiredProducer,
+            DockerStatusBuildItem dockerStatusBuildItem) {
+        if (ConfigUtils.isPropertyNonEmpty(OIDC_CLIENT_REG_AUTH_SERVER_URL_CONFIG_KEY)) {
```

Comment:
It is a bit tricky to check on my mobile phone, but it feels like some good portions of the code, for checking properties, docker, initial parts of the configurators can be pushed down to some common code for all 3 extensions... Have a look please,  LGTM overall

Reply:
I extracted code "duplicates" into the Keycloak Dev Service extension. It require few loops, I think I got them right, but I'll need to need to re-test it with Quarkus DEV mode and different extension combinations.

Reply:
Done. I have fixed the issue and retested it.

side note: I am starting to think I should always wait for the morning before I push changes :-)